src/sagemaker/predictor.py

```diff
@@ -94,6 +86,24 @@ def predict(self, data):
         response_body.close()
         return data
 
+    def _create_request_args(self, data, initial_args=None):
+        args = dict(initial_args) if initial_args else {}
+
+        if 'EndpointName' not in args:
+            args['EndpointName'] = self.endpoint
+
+        if self.content_type and 'ContentType' not in args:
+            args['ContentType'] = self.content_type
+
+        if self.accept and 'Accept' not in args:
+            args['Accept'] = self.accept
+
+        if self.serializer is not None:
+            data = self.serializer(data)
+
+        args['Body'] = data
+        return args
```

Comment:
nit: You can simplify the code here:

args['ContentType'] = args.get('ContentType', self.content_type)
args['Accept'] = args.get('Accept', self.accept)
args['Body'] = self. serializer(data) # we can provide a default serializer that just return the data itself.

Reply:
nicer, but behavior is diff't because it misses the check for self.content_type etc.
this is mostly pre-existing code (just moved) and i don't want to alter the behavior

Reply:
It's ok if you don't want to change but
```args['ContentType'] = args.get('ContentType', self.content_type)```
and
```
if self.content_type and 'ContentType' not in args:
    args['ContentType'] = self.content_type
```
have the same behavior.

Reply:
No, not quite.

In the former, you're going to always assign args['ContentType'] a value, which can be None.

Reply:
That is correct, my suggestion changes behavior. Thanks for clarifying.